bids-validator/validators/microscopy/validateTiffSignature.js

```diff
@@ -0,0 +1,20 @@
+import isNode from '../../utils/isNode'
+
+const getDataView = buffer => {
+  if (isNode) {
+    const uint8arr = new Uint8Array(buffer.byteLength)
+    buffer.copy(uint8arr, 0, 0, buffer.byteLength)
+    return new DataView(uint8arr.buffer)
+  } else {
+    return new DataView(buffer)
+  }
+}
+
+const validateTiffSignature = (buffer, tiffId) => {
+  const dataView = getDataView(buffer)
+  const littleEndian = dataView.getUint16(0) === 0x4949
+
+  return dataView.getUint16(2, littleEndian) === tiffId
```

Comment:
Just to be sure, this will check if the byte order is littleEndian or bigEndian and check the version number at byte 2 or 3 accordingdly?

Reply:
Yes.